File: backend/types/effect.go
package types

type Effect struct {
	AmountPerTurn float32 `yaml:"AmountPerTurn"`
}

type UserSkillEffect struct {
	Effect      `yaml:",inline"`
	UserSkillID `yaml:"SkillId"`
}

type UserStatEffect struct {
	Effect     `yaml:",inline"`
	UserStatID `yaml:"StatId"`
}

type TerritoryStatEffect struct {
	Effect          `yaml:",inline"`
	TerritoryStatID `yaml:"TerritoryStatId"`
}

type ZoneStatEffect struct {
	Effect     `yaml:",inline"`
	ZoneStatID `yaml:"ZoneStatId"`
}

File: frontend/src/types/effect.ts
export interface Effect {
  EffectId: string;
  AmountPerTurn: number;
}

export interface UserSkillEffect extends Effect {
  SkillId: string;
}

export interface UserStatEffect extends Effect {
  StatId: string;
}

export interface TerritoryStatEffect extends Effect {
  TerritoryStatId: string;
}

export interface ZoneStatEffect extends Effect {
  ZoneStatId: string;
}
